crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -1307,15 +1346,16 @@ pub struct EntityLocation {
     pub table_row: TableRow,
 }
 
-/// An [`Entity`] id may or may not correspond to a valid conceptual entity.
-/// If it does, the conceptual entity may or may not have a location.
-/// If it has no location, the [`EntityLocation`] will be `None`.
+/// An [`EntityRow`] id may or may not currently be spawned.
+/// If it is not spawned, the [`EntityLocation`] will be `None`.
 /// An location of `None` means the entity effectively does not exist; it has an id, but is not participating in the ECS.
 /// This is different from a location in the empty archetype, which is participating (queryable, etc) but just happens to have no components.
+/// For more information about what a `None` location means, see the module [docs](crate::entity).
 ///
 /// Setting a location to `None` is often helpful when you want to destruct an entity or yank it from the ECS without allowing another system to reuse the id for something else.
 /// It is also useful for reserving an id; commands will often allocate an `Entity` but not provide it a location until the command is applied.
-pub type EntityIdLocation = Option<EntityLocation>;
+/// For more information about these more complex entity life cycles, see the module [docs](crate::entity).
+pub type EntityRowLocation = Option<EntityLocation>;
```

Comment:
I kind of want to just remove this type entirely in favor of `Option<EntityLocation>`. I don't think this alias is improving the legibility / grokability of our codebase.